### DISCUSSION TITLE
Respecting different play styles

### DISCUSSION THREAD

**Discussion by dahe-at-google** [endless-sky/endless-sky#7827] [Balance, Mechanics, and Interactions]
Is Endless Sky a combat game, a trading game or a narrative game?

Yes!

It is all of these, and the reason I enjoy it is that it finds a balance between the three, leading to different styles of play. You can't completely neglect any one aspect, but you can use each of them to shift the emphasis to the others. I like the narrative aspect of working through the plot lines best, so I tend to start by using trading (missions, for example) to build lots of cash; use cash and capturing to obtain lots of ships; and use the ships to make combat relatively easy so that I can focus on my progress through the narrative.

I worry a bit that the developer community places a bit too much emphasis on enhancing and adding complication to the combat aspect, and more generally to the gear and tactics side of the game. That's certainly the impression I get from watching the discord channels, seeing many of the plugins, and reading proposals such as https://github.com/endless-sky/endless-sky/issues/6986 or https://github.com/endless-sky/endless-sky/issues/7333 (not to pick on anyone, just a couple that I saw on a quick scan of the issues list). Neither would stop me playing the game, but they would detract from the experience. I don't mind working out a balance between heat, energy and the features of my ships. The 6986 proposal says "Capping gets much harder/more taxing/more specialised in general" and lists a whole lot more parameters I would need to understand instead of just loading up with rifles and clicking Attack.  I don't want it harder or more taxing. That's not fun for me.

How to address this? First, respect that there are different play style and people enjoy different things about the game. Then consider making complex dynamics controllable through preferences. I know adding preferences can pretty easily get out of control and they make testing and balance harder, but it could be made manageable by having a small number of pre-defined play styles each mapping to a collection of different code paths. For example, "tactician" gets you more of the detailed combat features; "explorer" simplifies combat; "accountant" means the logic of purchasing and depreciation is made more complex (I mean this one a bit facetiously). If you want an in-game mode, allow me to purchase a tactical computer which will allow my captain to just yell "attack" and does all the tricky stuff without taking me through a more complex UI.

I have a couple of further idea which carry this through into the narrative structure, and aim to write them up when I have a little more time. For now my request is along the lines of imagine different players as you add features.

**Comment by Amazinite** [comment]
> First, respect that there are different play style and people enjoy different things about the game. Then consider making complex dynamics controllable through preferences.

That's the plan: #5556.

**Comment by Saugia** [comment]
I will say, regardless if it's a matter of making combat harder, or returns less beneficial, capturing is large and by far the easiest and best way to make money in the game, and is most certainly not comparable to other aspects. I'm not sure about making it more complex, but I definitely think there should be more diminishing returns. Compared to trading, or story play, capturing is 100+% better and very easy at giving the player income and ships to combat with and there are no downsides to it whatsoever since it's simply a ladder someone is able to climb.

What I'm basically saying is, all the talk on reworking capturing is warranted because it currently stands out from other playstyles. When we talk about editing it, we are talking about bringing it down to be more fair with them, not make it worse or harder. More dynamic, sure. But at the end of the day, it's easily the most done option, the easiest option, and the option with no downsides. It also opens a new issue when it comes to having massive fleets at points in the story we don't want that to happen at, but that's a different (but important to keep in mind) tangent.

We definitely do want to bring more attention to other playstyles, but capturing needs a look-at regardless, and that look-at shouldn't be simply avoided by throwing it into gamerules.

**Comment by EjoThims** [comment]
I'm not understanding how balancing and refining the capping meta has any impact on the ability to enjoy other play styles.

**Reply by Amazinite** [reply]
Looking at both issues mentioned, I think the concern is moreso a matter of unnecessary complexity in the suggested changes, not the act of balancing/refining itself.

**Reply by EjoThims** [reply]
That would be the refinement part.

But still, how does increased complexity in one play style disrespect the styles that don't rely on it anyway?

I'd understand if it was about amount of attention, but title and complaint here just seem wholly disconnected to me.

**Reply by dahe-at-google** [reply]
> But still, how does increased complexity in one play style disrespect the styles that don't rely on it anyway?

Because every play style in some way depends on every mechanic in the game to some degree. So if you force complexity in one area, either players have to work with that complexity, or they have to work round it. To be clear, capturing was just an example that came to hand, but to continue it, if you make it harder to use as a way of building up combat strength, you are forcing players down a different route, such as grinding trade rather than giving them a choice. Balance is good, but it's a poorly defined and relative notion, and what looks like balance to (say) a dev who enjoys combat may not seem like balance to someone else.

**Reply by EjoThims** [reply]
The choice is still there. That the player makes it based on their play preferences does not take said choice from them. No one is forced to capped ships to fund their merchant expansion, and no one will be forced not to after the changes to it, regardless of whether they personally enjoy those changes or not.

Moreover, the fact that capping is SEEN as necessary for even pacifist play styles because it is such a better source of both ships and income clearly means it is too powerful and influential, and the two should be -able- to be disconnected.

**Reply by MasterOfGrey** [reply]
Every playstyle depends a little bit on trade.
The same on combat.
The same narrative and jobs management.

I don't think you can actually say that all play styles depend on capturing in some way or form though.
Though as Ejo makes the valid point - if it's seen as necessary that's kind of a problem, since it is not one of the core 3 elements.

One point worth raising though, is that a certain amount of complexity is necessary for a system to be *able* to be balanced.
Combat has many factors, speed, acceleration, weapon damage types, hull/shield ratios, energy regen vs battery, the whole outfitting shebang, player skill at the controls etc.
Trading has at least several factors, balancing the price margins with the number of jumps, the size of your crews, and the price change generated by large and frequent trades.
And following the narratives have as many factors as the authors write in.

However, right now capturing works on a very simple formula, where whatever weapon has the most attack strength and the least defence strength is the best equipment to use 100% of the time, no exceptions.
It lacks a necessary dimension to make it balanced - because it lacks a necessary dimension for balance to even be *possible*.

So, I think it's a rather poor pick to make your case on.

Outfit durability and repairs would add complexity to the outfitting and combat elements which are already complex in multi-faceted ways.

Capturing mechanics are 1 dimensional.

To balance the capturing system, additional complexity is *necessary*.

However, in reference to 7333, and other things. I want to say that in principle I do actually agree with you, and have argued against additional complexity in some parts of the game from time to time.
There are many ways to play the game, and there's a certain degree of simplicity, or at least of streamlined experience, that gives it its charm. 
Complexity for it's own sake doesn't necessarily serve to enhance the game.

**Comment by Zitchas** [comment]
While I agree that combat gets a *lot* of focus, as a primarily story-driven content creator I'd like to point out that we've made massive strides lately (and have a bunch more in the works) to make more varied stories possible:
- Autoconditions (that let us check for and respond to things like what kind of ship someone is flying or what their stats are, or even how many ships they have.)
- Treating outfitters and shipyards like the spaceport (so we can have missions trigger in these locations instead of having everything being limited to spaceport, landing, or job board).
- Adding the ability to have missions to check for and respond to a wider variety of actions (including provoking without killing)
- Global variables that we could use to break the fourth wall if we wanted, implement alternate starts that otherwise aren't possible because their very name would be a spoiler, etc.

And lots more. We haven't even begun to scratch the surface of the improvements to storytelling that most of these have (or will) unlock for us.

As far as mechanics goes, yes, capturing does need a little more complexity. Not a lot, but enough that it isn't a purely one-dimensional thing. It is *not* a necessary thing. In fact, it is probably one of the areas of the game that is the *least* necessary, second only to tributing planets. And capturing does have a finite upper limit. It's rather like mining, honestly: It is *very* profitable, especially in the early to mid game, and it is the only way to get some ships, so it has a strong draw even in the late game. But in terms of profits it does have a certain plateau because, with the exception of capturing automata, one can only do so much capturing per run.

Compare this to commodity trading, which while slightly more complex in having crew costs, pirate attraction (in human space, anyway), and managing a fleet, also has the capacity to scale more or less infinitely. Only limited by the capacity of one's computer. I know I can get to 1000+ freighters in my fleet, raking in *billions* of credits in mere minutes. Forget dominating planets, when the game has to shift into scientific notation to display one's account balance, I should be able to *buy* planets. At least the small, underpopulated ones, anyway.

Your mention of the "tactical computer" raises a good point. This is something that I've been advocating for a while. As a case in point, fuel: The games that ES is based on / inspired by had fuel much the same way as we do - except it was an actual resources: One had to  pay for fuel at a fairly significant rate. Something like 1000 credits per hyperjump's worth of fuel. And when you land, there was a "refuel" button that you had to remember to hit. Or choose to hit. In the early game, sometimes one *didn't* want to refuel because you needed that money to buy commodities to sell in the next system, and you already had the fuel to make it. And if you run out in space, sure, lots of people will help you out... for a price. And they'd generally charge double, so 2000 credits per hyperjump worth of fuel. However, there was an outfit one could purchase, right from the beginning. One probably couldn't *afford* it until one had made a moderate amount of money, but it was called the auto-refueller, and it did exactly that. It automatically refuelled the ship. One still had to pay for the fuel, just it removed the complexity. In fact, it was basically an achievement: You've made enough money that you can buy an otherwise useless outfit just to save yourself the hassel of clicking a refuel button. It was, quite literally, a display of wealth.

MZ didn't like that complexity, though, so he just wrote it out. Fuel is free, ships automatically refuel, and ships in space will always give you a jump worth of fuel (unless they don't have any or doing so would run themselves out of fuel), and ships even automatically regenerate it naturally while just floating in space. 

This is a prime case where someone was reactionary against complexity and over-simplified. (There's a bunch of other examples with scanners and other stuff, too)

Anyway, the point being that I rather like having outfits that change the way things works. Having an outfit that one can buy that will automatically manage boarding combat is a good idea. I personally think that the auto-aim that we can turn on in the preferences should actually be an outfit that we choose to install, not a setting. Automatic landing and auto navigating along planned hyperjumps I think should *also* be outfits that one adds onto the ship; not built-in things. It lets the people who enjoy those complexities have them, while giving people a sense of accomplishment of getting rid of them if they want. Letting people *earn* stuff instead of just giving it to them generally leads to happier players, and more engaged players. (this is a hallmark of MMORPGs: Make people work to earn conveniences, aesthetic options, etc. Don't just let them have them)

**Comment by SimulatedKnave** [comment]
Speaking as a new player, I think capturing should probably be toned down a bit. But that if it's so obviously what everyone goes for, that is a sign the other options probably need to be improved a bit, not capturing nerfed. 

Let me make thousands of credits selling small cargos of luxury goods from planet to planet, for example. Right now, the big price swings on them are at most a few hundred credits. If you've got a shuttle, finding cheap luxury goods and selling them high SHOULD be a viable way to make money, and it doesn't really seem like it is. Sure, you can make piles of money trading by having a huge cargo capacity. You can make piles of money doing ANYTHING by throwing enough ships at it. 

Or let me book large groups of passengers, in the dozens, for short jumps. The game's never offered me the option to take 30 people to the next planet over. So if I buy a ship with a big passenger capacity, it's getting used for space marines and building a fleet, because reliable passenger transport isn't practical (and because the game throws pirate ships at me constantly, and damned if I'm gonna blow them all up. They're worth MONEY). 

Honestly, there's kind of a shortage of local missions generally. Instead everyone wants to go to the other side of the galaxy. 

My own thought on capturing elsewhere was adding prize mechanics - you have to pay to keep the ship, otherwise next time you land it's sold off at the bank and the money distributed amongst your crew (you get a big chunk, obviously). Makes it less insanely lucrative, while still rewarding, and explains why your faceless crewmembers are willing to throw themselves into the meatgrinder. If it's a bit less insanely lucrative, it's suddenly a lot more balanced (and if the ships are converted into money unless you keep them, you have less of a tendency to accidentally snowball into having a twenty-strong fleet). 

PS: And auto-aim shouldn't be an outfit until the damn thing actually works all the time. I end up manually aiming WAY too often with it on.

**Comment by IsocratesGrey** [comment]
I agree. I'm very much a narrative focused player. It's why I've been working on a narrative structure that people can use to build new campaigns easier #7810 . Combat complexity isn't something I care about unless it improves the narrative. Something to consider when developing new things isn't just "is this fun" but "does this help the player tell the story they want to tell" And for a lot of players that story is "space warlord" and that's fair. But I hope that the amazing creators here keep asking themselves that question about how an addition helps the player tell their story.

**Comment by movingpictures** [comment]
" Compared to trading, or story play, capturing is 100+% better and very easy at giving the player income and ships to combat with and there are no downsides to it whatsoever since it's simply a ladder someone is able to climb."

Capturing invariably leads to losses. It's possible to squander hundreds - if not thousands - of lives, without ANY repercussions whatsoever. 

Solution: At some point, the deaths are going to make it harder to recruit new hires. Possible ways to deal with this:

1) The more bodies are killed, the more the reputation falters with various species. Currently, I can load up a Kimek Spire in Remnant space and capture Korath ships. Y'think the Remant are going to be OK with First Last throwing their children into a meatgrinder for such purposes?
2) After so much bloodshed, it's should cost a LOT more to hire extra grunts. Ratio the salary for extra grunts based on the death suffered by those who have gone before.
3) At some point, the word will get out that hiring on as a grunt to this First Last character is a fast way to end up dead. Think of a message, "You put the call out for mercenaries, but instead of the 57 you wanted, only eight sign up."

**Reply by dahe-at-google** [reply]
I think you missed the point of my original message.

**Reply by movingpictures** [reply]
Then you didn't take the time to understand my comment. 

The flow of conversation wasn't that complex: someone essentially agreed with your point, noting that the game engine rewards capturing vessels (ie: combat). I suggested some ways of making vessel boarding more punitive.

**Comment by yafflehk** [comment]
Capturing is way too easy and attractive as a way of getting rich and powerful. It should require way more effort and specialized equipment to either take outfits from a disabled ship or capture it and get it back to port. One idea I had was to have separate grades of outfitter/shipyard:
Major shipyard - rare, only populous planets will have these, they can repair any damage and make a captured ship fully functional again.
Normal shipyard - less rare, most planets will have these, they can repair your ship but can't help you with your capture.
Small shipyard - frontier worlds, you can get fuel and a shield recharge, but nothing else.

This should be combined with a price for cleaning and repairing a captured ship based on it's original cost and it's state after capturing - For instance, Ships that have been captured using illegal weapons would need to be cleaned and repaired at shipyards on pirate planets, they should be confiscated and the player fined if they attempt to repair them on a "civilized" world, even if they don't have the outfit on their flagship.
A grenade capture should cost more to repair than a laser rifle capture, if you used alien weaponry the price should go through the roof because "Who knows what horrible stuff my guys will have to touch", or "The yard master insists his people wear full protective gear for this job, the price just tripled".

Captured ships should be crippled - the current game shows that by making them hard to pilot - but they shouldn't be able to shoot either, maybe nine times out of ten the hyperdrive is destroyed and they can't leave the system at all, shields should only recharge up to 50%, and so on. Capturing should be doable, but it should be HARD.
